File: src/main/java/br/com/kleberson2santos/casadocodigov1/novoautor/Autor.java
package br.com.kleberson2santos.casadocodigov1.novoautor;

import org.springframework.util.Assert;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;
import java.time.LocalDateTime;

@Entity
public class Autor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private String email;
    private String descricao;
    private LocalDateTime instanteCriacao = LocalDateTime.now();

    @Deprecated
    public Autor(){
    }

    public Autor(@NotBlank String nome, @NotBlank @Email String email, @NotBlank @Size(max = 400) String descricao) {
        Assert.hasLength(nome, "O nome é obrigatório");

        this.nome = nome;
        this.email = email;
        this.descricao = descricao;
    }

    @Override
    public String toString() {
        return "Autor{" +
                "id=" + id +
                ", nome='" + nome + '\'' +
                ", email='" + email + '\'' +
                ", descricao='" + descricao + '\'' +
                ", instanteCriacao=" + instanteCriacao +
                '}';
    }
}

File: src/main/kotlin/com/dev/eficiente/casa/codigo/Autor.kt
package com.dev.eficiente.casa.codigo

import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import javax.persistence.*

@Entity(name = "autor")
data class Autor(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long,
    @Column(name = "nome")
    val nome: String,
    @Column(name = "descricao")
    val descricao: String,
    @Column(name = "email")
    val email: String,
    @UpdateTimestamp
    @Column(name = "data_adicionada")
    val dataAdicionada: Instant)
